packages/shared/lib/src/sdk/mod.rs

```diff
@@ -102,9 +120,87 @@ impl Sdk {
     ) -> Result<(), JsError> {
         let args = tx::unbond_tx_args(tx_msg, password)?;
 
-        namada::ledger::tx::submit_unbond(&mut self.client, &mut self.wallet, args)
+        // TODO: Handle epoch?
+        let (mut tx, _, pk, _) =
+            namada::ledger::tx::build_unbond(&mut self.client, &mut self.wallet, args.clone())
+                .await
+                .map_err(|e| JsError::from(e))?;
+
+        signing::sign_tx(&mut self.wallet, &mut tx, &args.tx, &pk)
+            .await
+            .map_err(|e| JsError::from(e))?;
+
+        namada::ledger::tx::process_tx(&self.client, &mut self.wallet, &args.tx, tx)
+            .await
+            .map_err(|e| JsError::from(e))?;
+
+        Ok(())
+    }
+
+    /// Contruct transfer data for external signers, returns byte array
+    pub async fn build_transfer(&mut self, tx_msg: &[u8]) -> Result<JsValue, JsError> {
+        let args = tx::transfer_tx_args(tx_msg, None, None)?;
+
+        let transfer = namada::ledger::tx::build_transfer(
+            &self.client,
+            &mut self.wallet,
+            &mut self.shielded_ctx,
+            args.clone(),
+        )
+        .await
+        .map_err(|e| JsError::from(e))?;
+
+        let bytes = transfer.0.try_to_vec().map_err(|e| JsError::from(e))?;
+
+        to_js_result(bytes)
+    }
+
+    // Append signatures and return tx bytes
+    pub fn sign_tx(
+        &self,
+        tx_bytes: &[u8],
+        data_key: String,
+        code_key: String,
+        header_key: String,
+    ) -> Result<JsValue, JsError> {
+        let mut tx: Tx = Tx::try_from(tx_bytes).map_err(|e| JsError::from(e))?;
+        let data_secret = SecretKey::from_str(&data_key).map_err(|e| JsError::from(e))?;
+        let code_secret = SecretKey::from_str(&code_key).map_err(|e| JsError::from(e))?;
+        let header_secret = SecretKey::from_str(&header_key).map_err(|e| JsError::from(e))?;
```

Comment:
Also something to consider is moving those functions calls out of the sdk struct to another struct/function(that does not use wasm_bindgen). This way you do not have to map_err all the time, you can just map_err result of your new function.